File: app/src/main/java/igrek/todotree/service/tree/persistence/IndentedLine.java
package igrek.todotree.service.tree.persistence;


class IndentedLine {
	
	private String indentedLine;
	private int indentation;
	
	IndentedLine(String unindentedLine) {
		indentation = countIndentation(unindentedLine);
		indentedLine = unindentedLine.substring(indentation).trim();
	}
	
	public IndentedLine(String indentedLine, int indentation) {
		this.indentedLine = indentedLine;
		this.indentation = indentation;
	}
	
	public String getIndentedLine() {
		return indentedLine;
	}
	
	public int getIndentation() {
		return indentation;
	}
	
	private int countIndentation(String unindentedLine) {
		int indentation = 0;
		while (indentation < unindentedLine.length() && unindentedLine.charAt(indentation) == '\t') {
			indentation++;
		}
		return indentation;
	}
}

File: app/src/main/java/igrek/todotree/service/tree/persistence/IndentedLine.kt
package igrek.todotree.service.tree.persistence

internal class IndentedLine {
    var indentedLine: String
        private set
    var indentation: Int
        private set

    constructor(unindentedLine: String) {
        indentation = countIndentation(unindentedLine)
        indentedLine = unindentedLine.substring(indentation).trim { it <= ' ' }
    }

    constructor(indentedLine: String, indentation: Int) {
        this.indentedLine = indentedLine
        this.indentation = indentation
    }

    private fun countIndentation(unindentedLine: String): Int {
        var indentation = 0
        while (indentation < unindentedLine.length && unindentedLine[indentation] == '\t') {
            indentation++
        }
        return indentation
    }
}